CRM/Financial/BAO/FinancialTypeAccount.php

```diff
@@ -91,7 +91,10 @@ public static function add(&$params, &$ids = NULL) {
       $financialTypeAccount->find(TRUE);
     }
     $financialTypeAccount->copyValues($params);
-    self::validateRelationship($financialTypeAccount);
+    $valid = self::validateRelationship($financialTypeAccount);
+    if (!$valid) {
+      return FALSE;
```

Comment:
@jitendrapurohit the function we are calling here appears to throw an exception - which seems really appropriate for the BAO layer. I feel like the form layer should catch the exception & call CRM_Core_Session::statusBounce().